packages/kbn-test/src/auth/saml_auth.ts

```diff
@@ -102,6 +112,42 @@ export const createCloudSession = async (params: CreateSamlSessionParams) => {
     };
   };
 
+  let attemptsLeft = retryParams.attemptsCount;
+  while (attemptsLeft > 0) {
+    try {
+      sessionResponse = await axios.request(requestConfig(cloudLoginUrl));
+      if (sessionResponse?.status !== 200) {
+        throw new Error(
+          `Failed to create the new cloud session: 'POST ${cloudLoginUrl}' returned ${sessionResponse?.status}`
+        );
+      }
+      const token = sessionResponse?.data?.token as string;
+      if (!token) {
+        throw new Error(
+          `Failed to create the new cloud session: token is missing in response data\n${JSON.stringify(
+            sessionResponse?.data
```

Comment:
What's the "shape" of `sessionResponse?.data`?
Anything that shouldn't be shown in there?

Reply:
Nothing comes to my mind. This end point is publicly available and you call it without authentication, actually to fetch token that you will use for SAML authentication.

Reply:
Wait, I'm wrong. Obviously we have to use SAML credentials to get EC token. Let me check how to get only needed info

Reply:
Running it locally shortly,  it's building on my local right now

Reply:
I pushed 360957508dfe8d4908f5fca8e27f9adec731f39a

So far we don't get any user sensitive data in case of status != 200, but I redacted few keys just in case